src/include/distributed/placement_connection.h

```diff
@@ -23,7 +23,8 @@ extern MultiConnection * StartPlacementConnection(uint32 flags,
 												  const char *userName);
 
 extern void ResetPlacementConnectionManagement(void);
-extern void CheckForFailedPlacements(bool preCommit, bool using2PC);
+extern void MarkFailedShardPlacements(void);
+extern void PostCommitMarkFailedShardPlacements(bool using2PC);
```

Comment:
These two functions behave very differently. Making them one function with a parameter to choose between behaviour A and behaviour B is just confusing.